### PR TITLE
bleachbit: 2.0 -> 2.2

### PR DESCRIPTION
**Pull request by dtzWill**
###### Motivation for this change

https://www.bleachbit.org/news/bleachbit-22

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/misc/bleachbit/default.nix [hunk 1]
```diff
@@ -1,25 +1,26 @@
...
 
   src = fetchurl {
-    url = "mirror://sourceforge/bleachbit/${name}.tar.bz2";
-    sha256 = "0ps98zx4n13q92bq7ykqi6hj3i7brdqgm87i9gk6ibvljp1vxdz9";
+    url = "mirror://sourceforge/${pname}/${pname}-${version}.tar.bz2";
+    sha256 = "1yj9bc3k6s1aib7znb79h5rybfv691zz4szxkwf9fm9nr0dws603";
   };
 
+  nativeBuildInputs = [ gettext ];
   buildInputs = [  pythonPackages.wrapPython ];
```
Comment (worldofpeace) [inline pkgs/applications/misc/bleachbit/default.nix]:
```suggestion
```

Unneeded as it's done in `mk-python-derivation.nix`.

File: pkgs/applications/misc/bleachbit/default.nix [hunk 2]
```diff
@@ -1,25 +1,26 @@
...
+
   doCheck = false;
 
   postInstall = ''
-    mkdir -p $out/bin
-    cp bleachbit.py $out/bin/bleachbit
-    chmod +x $out/bin/bleachbit
-
-    substituteInPlace $out/bin/bleachbit --replace "#!/usr/bin/env python" "#!${pythonPackages.python.interpreter}"
+    make install SHELL=${stdenv.shell} prefix=${placeholder "out"}
```
Comment (worldofpeace) [inline pkgs/applications/misc/bleachbit/default.nix]: Can't you drop this entirely and just use `makeFlags`?

**Comment by dtzWill** [comment]
On Fri, 22 Mar 2019 19:08:20 -0700, worldofpeace <notifications@github.com> wrote:
> worldofpeace commented on this pull request.
> 
> 
> 
> >    };
>  
> +  nativeBuildInputs = [ gettext ];
>    buildInputs = [  pythonPackages.wrapPython ];
> 
> ```suggestion
> ```
> 
> Unneeded as it's done in `mk-python-derivation.nix`.

Hmm, curious.  If I remove it the build fails trying to run `msgfmt`.
Or does it work for you w/o the `gettext` input?

**Comment by dtzWill** [comment]
On Fri, 22 Mar 2019 19:09:29 -0700, worldofpeace <notifications@github.com> wrote:
> worldofpeace commented on this pull request.
> 
> 
> 
> >    doCheck = false;
>  
>    postInstall = ''
> -    mkdir -p $out/bin
> -    cp bleachbit.py $out/bin/bleachbit
> -    chmod +x $out/bin/bleachbit
> -
> -    substituteInPlace $out/bin/bleachbit --replace "#!/usr/bin/env python" "#!${pythonPackages.python.interpreter}"
> +    make install SHELL=${stdenv.shell} prefix=${placeholder "out"}
> 
> Can't you drop this entirely and just use `makeFlags`?

Not AFAICT? Normally yes (and wouldn't need to set SHELL ourselves) but
I don't see a way to do that here.

If that seems wrong or it works for you LMK haha :).

**Comment by worldofpeace** [comment]
@dtzWill I think my comment on the mailing list appears to be different than what I communicated.

For example I suggested to remove `wrapPython`not `gettext`. and said
"Unneeded as it's done in mk-python-derivation.nix.".

**Comment by worldofpeace** [comment]
> Not AFAICT? Normally yes (and wouldn't need to set SHELL ourselves) but
I don't see a way to do that here.

> If that seems wrong or it works for you LMK haha :).

Ahh realized that we should set `format` to other so it doesn't use the setup.py.
This should fix this I think
```diff
diff --git a/pkgs/applications/misc/bleachbit/default.nix b/pkgs/applications/misc/bleachbit/default.nix
index b2348bbf752..7b4b9cd0885 100644
--- a/pkgs/applications/misc/bleachbit/default.nix
+++ b/pkgs/applications/misc/bleachbit/default.nix
@@ -3,6 +3,8 @@ pythonPackages.buildPythonApplication rec {
   pname = "bleachbit";
   version = "2.2";
 
+  format = "other";
+
   src = fetchurl {
     url = "mirror://sourceforge/${pname}/${pname}-${version}.tar.bz2";
     sha256 = "1yj9bc3k6s1aib7znb79h5rybfv691zz4szxkwf9fm9nr0dws603";
@@ -17,11 +19,13 @@ pythonPackages.buildPythonApplication rec {
     find -type f -exec sed -i -e 's@/usr/bin@${placeholder "out"}/bin@g' {} \;
   '';
 
-  doCheck = false;
+  dontBuild = true;
 
-  postInstall = ''
-    make install SHELL=${stdenv.shell} prefix=${placeholder "out"}
-  '';
+  installFlags = [ "prefix=${placeholder "out"}" ];
+
+  doCheck = true;
+
+  checkTarget = "tests";
 
   propagatedBuildInputs = with pythonPackages; [ pygtk ];
 
```

**Comment by dtzWill** [comment]
On Fri, 22 Mar 2019 21:42:55 -0700, worldofpeace <notifications@github.com> wrote:
> @dtzWill I think my comment on the mailing list appears to be different than what I communicated.
> 
> For example I suggested to remove `wrapPython`not `gettext`. and said
> "Unneeded as it's done in mk-python-derivation.nix.".


...

Yes you are correct, I completely misread/misunderstood.

Sorry :3.  Pushed commit making the change, good call thanks! :)

**Comment by worldofpeace** [comment]
Don't see the tests running in the output, not sure why. Guess we should just leave that out so it doesn't look confusing.

**Comment by worldofpeace** [comment]
ping @dtzWill

**Comment by dtzWill** [comment]
Pushed commit disabling tests, sorry for delay.
Was hoping to fix them (moving to `installCheckPhase = ` gets them
running) but too many small details to fixup so punting on that for now.

**Comment by worldofpeace** [comment]
> Pushed commit disabling tests, sorry for delay. Was hoping to fix them (moving to `installCheckPhase = ` gets them running) but too many small details to fixup so punting on that for now.
> […](#)
> On Sun, 31 Mar 2019 21:56:31 -0700, worldofpeace ***@***.***> wrote: ping @dtzWill -- You are receiving this because you were mentioned. Reply to this email directly or view it on GitHub: [#58152 (comment)](https://github.com/NixOS/nixpkgs/pull/58152#issuecomment-478435625Non-text) part: text/html

Looks good. It just needs a squash. I'd try to maybe keep some of the comments on what was done.

Other than that this works :+1: .

**Pull request merged**